### PR TITLE
Node 16

### PR DESCRIPTION
**Pull request by solomon-negusse**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -24,7 +26,7 @@ COPY ./app /opt/$NAME/app
 RUN chown $USER:$USER /opt/$NAME
 
 # Tell Docker we are going to use this ports
-EXPOSE 4000
+EXPOSE 30521
```
Comment (tiagojsag) [inline Dockerfile:29]: This will break on AWS.

Comment (tiagojsag) [inline Dockerfile]: This will break AWS deployments

File: README.md [hunk 2]
```diff
@@ -54,7 +54,14 @@ The endpoints provided by this microservice should now be available through Cont
...
 
-2 - Execute the following command to run Control tower:
+2 - Create a docker network for services including elasticsearch and rabbitmq that are shared across document
+microservices:
+
+```
+docker network document-adapter
+```
+
+3 - Execute the following command to run Control tower:
```
Comment (tiagojsag) [inline README.md:64]: if you are reviewing the docs (which is something I would encourage, but perhaps as part of a separate PR) you should remove references to CT, which we are no longer using.

File: app/src/services/scroll.js [hunk 3]
```diff
@@ -183,9 +183,7 @@ class Scroll {
 
         if (this.resultScroll.aggregations) {
             const data = Scroll.serialize(this.resultScroll, this.parsed, this.format);
-            this.stream.write(this.convertDataToDownload(data, this.format, true, false, this.cloneUrl), {
```
Comment (tiagojsag) [inline app/src/services/scroll.js:186]: Functional, unrelated changes.

File: docker-compose-develop.yml [hunk 4]
```diff
@@ -3,19 +3,55 @@ services:
...
       NODE_ENV: dev
-      PORT: 4000
+      PORT: 30521
       NODE_PATH: app/src
       MICROSERVICE_TOKEN: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6Im1pY3Jvc2VydmljZSIsImNyZWF0ZWRBdCI6IjIwMTYtMDktMTQifQ.IRCIRm1nfIQTfda_Wb6Pg-341zhV8soAgzw7dd5HxxQ
-      STAMPERY_TOKEN:
-      S3_ACCESS_KEY_ID:
-      S3_SECRET_ACCESS_KEY:
-      ELASTIC_URL: http://localhost:9200
+      GATEWAY_URL: http://mymachine:9000
```
Comment (tiagojsag) [inline docker-compose-develop.yml:15]: same thing as I mentioned on a different PR: there are better ways to do this. This, as is,  will cause problems

**Review by tiagojsag — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (solomon-negusse) [reply README.md:64]: Yes, I do read and reread the docs and appreciate the detail but this line isn't a new addition to this doc. As it stands, CT is required for dev and I'll defer removing these references to when there is a working and documented alternative.

File: docker-compose-develop.yml (hunk 4 above)
Comment (solomon-negusse) [reply docker-compose-develop.yml:15]: Simply adding a missing environment variable here the way you have set them in the other MSs (e.g., dataset) to get this working. Can we do a system-wide change in a separate issue to address your concern?

Comment (tiagojsag) [reply docker-compose-develop.yml:15]: the concern is not the variable, is the value.

Comment (solomon-negusse) [reply docker-compose-develop.yml:15]: Understood, but that's the recommended approach in the docs (e.g., in [dataset](https://github.com/resource-watch/dataset/blob/dev/docker-compose-develop.yml#L13) you have the same value in docker-compose). I can take it out into the dev.env file that's not version controlled if this feels hard-coded.

Comment (tiagojsag) [reply docker-compose-develop.yml:15]: those instructions are old, and should be reviewed. AFAIK those are Mac/Windows specific, and as they currently stand, they are incomplete, as there are no docs explaining what that fake host is supposed to mean - without which they won't work. This is another reason why I would encourage you to focus these changes on the nodejs version upgrade, and leave everything else for separate PRs
